app/sequencescape/sequencescape/api/v2/shared/has_workline_identifier.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Sequencescape::Api::V2::Shared
+  # Include in API endpoints that have barcodes to add a few standard methods
+  module HasWorklineIdentifier
+    extend ActiveSupport::Concern
+
+    # Finds the labware whose barcode will be printed in the label at the top_right field.
+    # Uses alternative_workline_identifier from the purpose config if present, otherwise uses the stock plate.
+    def workline_reference
+      alternative_workline_identifier_purpose = SearchHelper.alternative_workline_reference_name(self)
+      return stock_plate if alternative_workline_identifier_purpose.nil?
+
+      ancestors.where(purpose_name: alternative_workline_identifier_purpose).last
+    end
+
+    def workline_identifier
+      workline_reference&.barcode&.human
+    end
+  end
+end
```

Comment:
This code seems to be extracted from plate so it can be included in other labware. The common use of the identifier is using it at the top-right field if it is a Plate barcode. Label templates delegate the workline_identifier method to their labware. One can still override it inside label template or labware if this concern is included. Making this a concern is a very good approach. For tubes though, the printed label can use it for one of the lines in the label. Maybe the comment of workline_reference of the Concern could reflect that.